src/framework/mpe/mpetypes.h

```diff
@@ -385,53 +395,16 @@ struct ArticulationPatternSegment
     ExpressionPattern expressionPattern;
 };
 
-using ArticulationPattern = std::map<duration_percentage_t, ArticulationPatternSegment>;
+using ArticulationPattern = SharedMap<duration_percentage_t, ArticulationPatternSegment>;
 
-struct ArticulationsProfile
+struct ArticulationsProfile : public SharedMap<ArticulationType, ArticulationPattern>
```

Comment:
Not a fan of this solution, I think the previous one was better.
There are several disadvantages when inheriting from standard components:
* The interface is too large, it may be that some methods are unnecessary or harmful.
* Too abstract interface (for example, `T pattern (...)` method is better than `T at (...)`)
* Difficult to debug - e.g. set breakpoint or log
* Difficult to change implementation

Reply:
Me either. I've changed it by analogy with ArticulationMap, in which I need almost the entire SharedHashMap interface and it is more convenient to use inheritance there.

However, now I see that it's unnecessary to use inheritance here. Returned the previous one. Thanks

Reply:
Yes, inheritance is fine for `ArticulationMap`, because it's a `map` :)